src/hypothesis/strategies.py

```diff
@@ -695,6 +695,32 @@ def text(
 
 @cacheable
 @defines_strategy
+def strings_matching_regex(regex):
+    """Return strategy that generates strings that match given regex.
+
+    Regex can be either a string or compiled regex (through
+    :func:`re.compile`).
+
+    You can use regex flags (such as :const:`python:re.IGNORECASE`,
+    :const:`python:re.DOTALL` or :const:`python:re.UNICODE`) to control
+    generation. Flags can be passed either in compiled regex (specify flags in
+    call to :func:`re.compile`) or inside pattern with (?iLmsux) group.
+
+    Some tricky regular expressions are partly supported or not supported at
+    all. "^" and "$" do not affect generation. Positive lookahead/lookbehind
+    groups are considered normal groups. Negative lookahead/lookbehind groups
+    do not do anything. Ternary regex groups
+    ('(?(name)yes-pattern|no-pattern)') generate variants regardless of
+    presence of named group and rely on filtering to produce correct results
+    (therefore can be slower than straightforward generation).
+
+    """
```

Comment:
Add a line of validation:

    if not isinstance(getattr(regex, 'pattern', regex), text_type):
        raise InvalidArgument(
            'regex=%s, but the strategy only supports unicode patterns.  '
            'Please open an issue if you need byte-string regexes' % (regex,))

This will also need a small test 😐